pkg/repo/chartrepo.go

```diff
@@ -198,13 +200,13 @@ func (r *ChartRepository) generateIndex() error {
 // FindChartInRepoURL finds chart in chart repository pointed by repoURL
 // without adding repo to repositories
 func FindChartInRepoURL(repoURL, chartName, chartVersion, certFile, keyFile, caFile string, getters getter.Providers) (string, error) {
-	return FindChartInAuthRepoURL(repoURL, "", "", chartName, chartVersion, certFile, keyFile, caFile, getters)
+	return FindChartInAuthRepoURL(repoURL, "", "", chartName, chartVersion, certFile, keyFile, caFile, "", getters)
 }
 
 // FindChartInAuthRepoURL finds chart in chart repository pointed by repoURL
 // without adding repo to repositories, like FindChartInRepoURL,
 // but it also receives credentials for the chart repository.
-func FindChartInAuthRepoURL(repoURL, username, password, chartName, chartVersion, certFile, keyFile, caFile string, getters getter.Providers) (string, error) {
+func FindChartInAuthRepoURL(repoURL, username, password, chartName, chartVersion, certFile, keyFile, caFile, token string, getters getter.Providers) (string, error) {
```

Comment:
Helm uses [semantic versioning](https://semver.org). This function is part of the public API. Changing the signature is considered a breaking change that would require incrementing the major version to Helm v4. We aren't going to do that right now.

Instead, you can have a new function for this change while preserving the old one. The old one can be marked as deprecated. #7984 is a PR that does this, as an example.

Reply:
Thanks, I have update my commit that follow your advise